cmds/ls/fileinfo_linux.go

```diff
@@ -34,10 +34,21 @@ type fileInfo struct {
 	uid, gid     uint32
 	size         int64
 	modTime      time.Time
+	symlink      string
 }
 
-func extractImportantParts(fi os.FileInfo) fileInfo {
+func extractImportantParts(n string, fi os.FileInfo) fileInfo {
+	var link string
+
 	s := fi.Sys().(*syscall.Stat_t)
+	if fi.Mode()&os.ModeType == os.ModeSymlink {
+		if l, err := os.Readlink(n); err != nil {
+			link = err.Error()
```

Comment:
This error message might look a bit strange when printed:

     c -> error file not found
     d -> permission denied

On the other hand, dropping the error might be worse.